### ISSUE TITLE
Adding Predictions does not do well

### ISSUE THREAD

**Issue by LogOg** [dotnet/docs#7029]
Added a few more tests and results not great
Sentiment Predictions
---------------------
Sentiment: Please refrain from adding nonsense to Wikipedia. | Prediction: Negat
ive
Sentiment: He is the best, and the article should say that. | Prediction: Positi
ve
Sentiment: You are just rude. | Prediction: Positive
Sentiment: Your staff left me standing for 15 minutes | Prediction: Positive
Sentiment: Great service, thank you! | Prediction: Positive
Sentiment: Why do I have to enter my data again, wastes my time | Prediction: Po
sitive

Then added few more rows to training set to try improve things with similar sentiment to above but came out same except this changed to Negative. 
Great service, thank you! | Prediction:
Is the training set just too small or perhaps the text samples too long.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a30b3ff5-b5e6-ff02-6f60-b5cef6f34666
* Version Independent ID: fdf962e2-620e-d9ad-5057-5f96f52dc60f
* Content: [Use ML.NET to predict New York taxi fares (regression)](https://docs.microsoft.com/en-us/dotnet/machine-learning/tutorials/taxi-fare#prepare-and-understand-the-data)
* Content Source: [docs/machine-learning/tutorials/taxi-fare.md](https://github.com/dotnet/docs/blob/master/docs/machine-learning/tutorials/taxi-fare.md)
* Product: **dotnet-ml**
* GitHub Login: @aditidugar
* Microsoft Alias: **johalex**

**Comment by mairaw** [comment]
Thanks for your feedback @LogOg. This seems a duplicate of #7029. @JRAlexander, what do you think?

**Comment by pkulikov** [comment]
It's related to #7024

**Comment by LogOg** [comment]
Hi Guys,

The scenario I described was run on 0.3.0 not 0.4.0 so the issue of #7029
is not the cause

Thx,

Logan

**Comment by JRAlexander** [comment]
We'll have to do some more investigation with the team but given the different versions,  two different issues.

**Comment by JRAlexander** [comment]
@sfilipi or @GalOshri - any thoughts?

**Comment by pkulikov** [comment]
@LogOg thank you for clarification. I thought you've tried to update the training set to investigate #7024 

By the way, this issue is related to the sentiment analysis tutorial (not the taxi fare one as metadata points out):
https://docs.microsoft.com/en-us/dotnet/machine-learning/tutorials/sentiment-analysis

**Comment by GalOshri** [comment]
My guess is that the results are largely due to the sample datasets being very small and the tree parameters also being small (`{ NumLeaves = 5, NumTrees = 5, MinDocumentsInLeafs = 2 }`).

Text classification scenarios generally require more data to train a good model (the training data in this sample has 250 examples). You would also want to modify the tree parameters based on the dataset. Using the defaults helps improve the accuracy and AUC in this case.

**Comment by LogOg** [comment]
Hi Guys,

As this is the first people will see when investigating ML. Net I think it
is worth getting better samples, You probably need more like 5000 samples
and rather shorter less obscure samples. The model can then really train
properly and would be great if the example allowed typing in sentiments for
prediction even in razor but adding or changing in code is also fine.

Thx,

Logan

**Comment by JRAlexander** [comment]
try updating the FastTreeBinaryClassifier() hyperparameters to:  NumLeaves = 50, NumTrees = 50, MinDocumentsInLeafs = 20. Closing as appears resolved.
